python/src/uagents/wallet_messaging.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+import base64
+import functools
+import logging
+from typing import List, Optional
+
+from babble import Client, Identity as BabbleIdentity
+from babble.client import Message as WalletMessage
+from cosmpy.aerial.wallet import LocalWallet
+from requests import HTTPError, JSONDecodeError
+
+from uagents.config import (
+    WALLET_MESSAGING_POLL_INTERVAL_SECONDS,
+    get_logger,
+)
+from uagents.context import Context, WalletMessageCallback
+from uagents.crypto import Identity
+
+
+class WalletMessagingClient:
+    def __init__(
+        self,
+        identity: Identity,
+        wallet: LocalWallet,
+        chain_id: str,
+        logger: Optional[logging.Logger] = None,
+    ):
+        delegate_pubkey = identity.pub_key
+        delegate_pubkey_b64 = base64.b64encode(bytes.fromhex(delegate_pubkey)).decode()
+        public_key = base64.b64decode(wallet.public_key().public_key).hex()
+        signed_bytes, signature = identity.sign_arbitrary(public_key.encode())
+        self._client = Client(
+            identity.address,
+            delegate_pubkey_b64,
+            signature,
+            signed_bytes,
+            BabbleIdentity(wallet.signer().private_key_bytes),
+            chain_id,
+        )
+        self._poll_interval = WALLET_MESSAGING_POLL_INTERVAL_SECONDS
+        self._logger = logger or get_logger("wallet_messaging")
+        self._message_queue = asyncio.Queue()
+        self._message_handlers: List[WalletMessageCallback] = []
+
+    def on_message(
```

Comment:
As there is no qualifier for the handler (e.g. like a model on which to act), a developer can create multiple handlers which will all trigger at the same time whenever a wallet message is incoming.

Just raising this here as we don't prevent a dev from doing stupid things we may want to either:
- append the function to the existing handler,
- overwrite the existing handler, or
- prevent the handler from being overwritten, thus forcing the dev to structure their code differently.

Reply:
Yes, I made the initial quick decision to allow multiple handlers to be registered for wallet messages and haven't really revisited this. Not sure I'm convinced that these options are better, but it's probably worth creating an issue to come back to this.

Reply:
https://github.com/fetchai/uAgents/issues/206